Tests/Common/Securities/Futures/FuturesExpiryFunctionsTests.cs

```diff
@@ -51,27 +228,136 @@ public void TestGoldExpiryDateFunction()
+        {
+            var jun2017 = Symbol.CreateFuture(QuantConnect.Securities.Futures.Currencies.CAD, Market.USA, new DateTime(2017, 6, 20));
+            var jun2017Func = FuturesExpiryFunctions.FuturesExpiryFunction(jun2017.ID.Symbol);
+            Assert.AreEqual(jun2017Func(jun2017.ID.Date), new DateTime(2017, 6, 20, 14, 16, 0));
+        }
+
+        [Test]
+        public void TestY5TreasuryNotesExpiryDateFunction()
+        {
+            var jun2017 = Symbol.CreateFuture(QuantConnect.Securities.Futures.Financials.Y5TreasuryNote, Market.USA, new DateTime(2017, 6, 30));
+            var jun2017Func = FuturesExpiryFunctions.FuturesExpiryFunction(jun2017.ID.Symbol);
+            Assert.AreEqual(jun2017Func(jun2017.ID.Date), new DateTime(2017, 6, 30, 12, 1, 0));
+        }
+
+        [Test]
+        public void TestCrudeOilWTINotesExpiryDateFunction()
+        {
+            var jun2017 = Symbol.CreateFuture(QuantConnect.Securities.Futures.Energies.CrudeOilWTI, Market.USA, new DateTime(2017, 6, 1));
+            var jun2017Func = FuturesExpiryFunctions.FuturesExpiryFunction(jun2017.ID.Symbol);
+            Assert.AreEqual(jun2017Func(jun2017.ID.Date), new DateTime(2017, 5, 22));
+        }
+
+        [Test]
+        public void TestHeatingOilNotesExpiryDateFunction()
+        {
+            var jun2017 = Symbol.CreateFuture(QuantConnect.Securities.Futures.Energies.HeatingOil, Market.USA, new DateTime(2017, 6, 1));
+            var jun2017Func = FuturesExpiryFunctions.FuturesExpiryFunction(jun2017.ID.Symbol);
+            Assert.AreEqual(jun2017Func(jun2017.ID.Date), new DateTime(2017, 5, 31));
+        }
+
+        [Test]
+        public void TestNaturalGasNotesExpiryDateFunction()
+        {
+            var jun2017 = Symbol.CreateFuture(QuantConnect.Securities.Futures.Energies.NaturalGas, Market.USA, new DateTime(2017, 6, 1));
+            var jun2017Func = FuturesExpiryFunctions.FuturesExpiryFunction(jun2017.ID.Symbol);
+            Assert.AreEqual(jun2017Func(jun2017.ID.Date), new DateTime(2017, 5, 26));
+        }
+
+        [Test]
+        public void TestLiveCattleExpiryDateFunction()
+        {
+            var jun2017 = Symbol.CreateFuture(QuantConnect.Securities.Futures.Meats.LiveCattle, Market.USA, new DateTime(2017, 6, 1));
+            var jun2017Func = FuturesExpiryFunctions.FuturesExpiryFunction(jun2017.ID.Symbol);
+            Assert.AreEqual(jun2017Func(jun2017.ID.Date), new DateTime(2017, 6, 30, 12, 0, 0));
+        }
+
+        [Test]
+        public void TestLeanHogsExpiryDateFunction()
+        {
+            var jun2017 = Symbol.CreateFuture(QuantConnect.Securities.Futures.Meats.LeanHogs, Market.USA, new DateTime(2017, 6, 1));
+            var jun2017Func = FuturesExpiryFunctions.FuturesExpiryFunction(jun2017.ID.Symbol);
+            Assert.AreEqual(jun2017Func(jun2017.ID.Date), new DateTime(2017, 6, 14, 12, 0, 0));
+        }
+
+        [Test]
+        public void TestFeederCattleExpiryDateFunction()
```

Comment:
There are several tests like this one that I believe are testing the futures contracts with more complex and custom expiry rules. It might be a good idea to add more than one date to test. You may be able to use Test Cases here too